libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -596,20 +596,24 @@ - (void)switchToNewUser {
 }
 
 - (void)switchToUser:(SFUserAccount *)newCurrentUser {
-    [self enumerateDelegates:^(id<SFUserAccountManagerDelegate> delegate) {
-        if ([delegate respondsToSelector:@selector(userAccountManager:willSwitchFromUser:toUser:)]) {
-            [delegate userAccountManager:self willSwitchFromUser:self.currentUser toUser:newCurrentUser];
-        }
-    }];
-
-    SFUserAccount *prevUser = self.currentUser;
-    [self setCurrentUser:newCurrentUser];
-
-    [self enumerateDelegates:^(id<SFUserAccountManagerDelegate> delegate) {
-        if ([delegate respondsToSelector:@selector(userAccountManager:didSwitchFromUser:toUser:)]) {
-            [delegate userAccountManager:self didSwitchFromUser:prevUser toUser:newCurrentUser];
-        }
-    }];
+    if (self.currentUser == newCurrentUser) {
```

Comment:
Shouldn't we use `isEqual` here? `isEqual` should compare if the `orgId`, `userId` and optionally `communityId` are the same to determine equality.